src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathDocumentBuilder.cs

```diff
@@ -882,10 +900,11 @@ public int LinePosition
             /// <summary>
             /// Append a text block with the specified type.
             /// </summary>
-            public void WriteTextBlock(string text, TextBlockType textType)
+            public void WriteTextBlock(string? text, TextBlockType textType)
             {
                 Debug.Assert((int)XPathNodeType.Text < (int)XPathNodeType.SignificantWhitespace);
                 Debug.Assert((int)XPathNodeType.SignificantWhitespace < (int)XPathNodeType.Whitespace);
+                Debug.Assert(text != null);
```

Comment:
why? both Concat and assignment seem fine, perhaps consider changing line 909 to `!string.IsNullOrEmpty`

Reply:
Many callers of `WriteString` can potentially pass `string?`, changing below condition to `!string.IsNullOrEmpty` sounds good to me, thanks.